test/ros_helpers_test.cpp

```diff
@@ -0,0 +1,37 @@
+#include <common_robotics_utilities/ros_helpers.hpp>
+
+#include <gtest/gtest.h>
+
+namespace common_robotics_utilities
+{
+namespace ros_helpers
+{
+namespace
+{
+
+#if COMMON_ROBOTICS_UTILITIES__SUPPORTED_ROS_VERSION == 2
+using Duration = rclcpp::Duration;
+#elif COMMON_ROBOTICS_UTILITIES__SUPPORTED_ROS_VERSION == 1
+using Duration = ros::Duration;
+#endif
+
+GTEST_TEST(ROSHelpersTest, SetMessageTimestamps)
+{
+  Marker marker;
+  MarkerArray marker_array;
+  marker_array.markers.push_back(marker);
+  Time base_stamp = marker.header.stamp;  // handle clock types (indirectly)
```

Comment:
I'd prefer this be a little different
```
VisualizationMarkerArray marker_array;
marker_array.markers.resize(5);

// Note that the ROS time type depends on which ROS variant is in use.
const RosTime base_stamp;
// Advance the timestamp by one second.
const RosTime new_stamp = base_stamp + RosDuration(1, 0);

SetMessageTimestamps(marker_array, new_stamp);

for (const auto& marker : marker_array.markers)
{
  EXPECT_EQ(marker.stamp, new_stamp);
}
```

Reply:
Sure. It'll not be just like that, because a default constructed `rclcpp::Time` cannot be compared with an implicitly cast `builtin_interfaces::msg::Time` stamp -- their clock sources do not match. Getting the base stamp from the marker ensures clock sources match in a portable way.

See https://github.com/calderpg/common_robotics_utilities/pull/9/commits/ee583a1ce6e6c252532fe5f1b7cd6f66f8c318f1.